node_scripts/job_submission.py

```diff
@@ -1,13 +1,22 @@
-import sys
+import datetime
 import os
-import yaml
 import subprocess
-import datetime
+import sys
 from typing import List
-import azure.storage.blob as blob
 import azure.batch.models as batch_models
+import azure.storage.blob as blob
+import yaml
 from command_builder import CommandBuilder
 from core import config
+from install.pick_master import get_master_node_id
+
+
+def affinitize_task_to_master(batch_client, cluster_id, task):
```

Comment:
I know we've talked about this before, but it is still strange to me that this code is essentially duplicated.

Reply:
yeah -- definitely not ideal, but this is one of the only options until we do the larger refactor we've discussed around loading the entire SDK (not just node scripts) as a resource file for the cluster. That refactor should simplify/eliminate a lot of the duplicate code we have in node scripts.